### PR TITLE
[Jobs] Wizard: preload parameters, data inputs and resources of the s…

### PR DESCRIPTION
**Pull request by IvanVoznyi**
…elected function

### CODE REVIEW

File: src/elements/JobsPanelTableRow/JobsPanelTableRow.js [hunk 1]
```diff
@@ -5,25 +5,50 @@ import Tooltip from '../../common/Tooltip/Tooltip'
...
+
+import { joinDataOfArrayOrObject } from '../../utils'
+
 import './jobsPanelTableRow.scss'
 
 const JobsPanelTableRow = ({ actionsMenu, item, row }) => {
   return (
     <div className="table__row">
-      {row.map((cell, i) => {
+      {Object.keys(row).map((property, i) => {
```
Comment (mavdryk) [inline src/elements/JobsPanelTableRow/JobsPanelTableRow.js:17]: you don't need `Object.keys` here

File: src/utils/string.js [hunk 2]
```diff
@@ -1,2 +1,11 @@
 export const truncateUid = (value = '') =>
   value.length > 7 ? `...${value.slice(-7)}` : value
+
+export const joinDataOfArrayOrObject = (data, splitCharacter = ',') => {
+  if (Array.isArray(data)) {
+    return data.join(splitCharacter)
+  } else if (typeof data === 'object') {
```
Comment (mavdryk) [inline src/utils/string.js:7]: use `lodash.isObject`

File: src/utils/object.js [hunk 3]
```diff
@@ -1,2 +1,12 @@
...
   Object.entries(object).map(([key, value]) => `${key}: ${value}`)
+
+export const parseJobsDefaultParameters = (parameters, method) => {
+  let parseParameters = parameters
+    .map(parameter => parameter.items)
+    .reduce((prev, curr) => {
+      let name = curr.name
+      return { ...prev, [name]: curr.value }
+    }, {})
+  method(parseParameters)
```
Comment (mavdryk) [inline src/utils/object.js:11]: you don't need `map` here, you can use `curr.items.name` and `curr.items.value` in the reduce method

File: src/utils/object.js [hunk 4]
```diff
@@ -1,2 +1,12 @@
 export const parseKeyValues = (object = {}) =>
   Object.entries(object).map(([key, value]) => `${key}: ${value}`)
+
+export const parseJobsDefaultParameters = (parameters, method) => {
+  let parseParameters = parameters
+    .map(parameter => parameter.items)
+    .reduce((prev, curr) => {
+      let name = curr.name
```
Comment (mavdryk) [inline src/utils/object.js:8]: delete `name` variable

File: src/utils/string.js [hunk 5]
```diff
@@ -1,2 +1,11 @@
...
+    return Object.values(data).join(splitCharacter)
+  }
+  return data.toString()
```
Comment (mavdryk) [inline src/utils/string.js:10]: empty line before `return`

File: src/components/JobsPanel/jobsPanel.scss [hunk 6]
```diff
@@ -226,5 +218,12 @@
 
   .no-border {
     border-bottom: none;
+
+    .select {
```
Comment (mariana-furyk) [inline src/components/JobsPanel/jobsPanel.scss:222]: your styles effects on select in edit mode, if you try to edit some parameter and click to add some new item in parallel

File: src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js [hunk 7]
```diff
@@ -63,7 +63,7 @@ const JobsPanelDataInputsView = ({
           setSelectedDataInput={setSelectedDataInput}
         >
           {addNewInput ? (
-            <>
+            <div className="table__row-create-data-input">
```
Comment (mariana-furyk) [inline src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js:66]: if you add this class, and other class on parameters just for flex, add to both one common class

File: src/components/JobsPanelParameters/JobsPanelParametersView.js [hunk 8]
```diff
@@ -46,14 +46,14 @@ const JobsPanelParametersView = ({
...
-            <>
+            <div className="table__row-create-parameter">
               <div className="input-row-wrapper">
                 <Input
                   onChange={value =>
                     setNewParameter({ ...newParameter, name: value })
                   }
                   label="Name"
-                  className="input-row__item"
+                  className="input-row__item parameter-name"
```
Comment (mariana-furyk) [inline src/components/JobsPanelParameters/JobsPanelParametersView.js:56]: if you add this classes just to add some padding to first element (and in other components too), just add some style to first child

File: src/elements/EditableParametersRow/EditableParametersRow.js [hunk 9]
```diff
@@ -15,29 +15,46 @@ const EditableParametersRow = ({
...
   return (
     <div className="table__row edit-row">
-      <div className="table__cell">{selectedParameter.name}</div>
-      <div className="table__cell">{selectedParameter.type}</div>
-      <div className="table__cell">
+      <div className="table__cell parameter-name">
+        {selectedParameter.items.name}
+      </div>
+      <div className="table__cell">{selectedParameter.items.type}</div>
+      <div className="table__cell table__cell_edit">
```
Comment (mariana-furyk) [inline src/elements/EditableParametersRow/EditableParametersRow.js:22]: you don't need this class and styles, it breaks view

File: src/elements/EditableVolumesRow/EditableVolumesRow.js [hunk 10]
```diff
@@ -15,22 +15,30 @@ const EditableVolumesRow = ({
...
+                ...selectedVolume,
+                items: { ...selectedVolume.items, mountPath: path }
+              })
             }
             type="text"
-            value={selectedVolume.mountPath}
+            value={selectedVolume.items.mountPath}
           />
         </div>
+        <div className="actions_cell"></div>
```
Comment (mariana-furyk) [inline src/elements/EditableVolumesRow/EditableVolumesRow.js:35]: rename this class, cell is not modificator

File: src/elements/EditableVolumesRow/editableVolumesRow.scss [hunk 11]
```diff
@@ -14,5 +13,30 @@
         }
       }
     }
+
+    .resource-path {
```
Comment (mariana-furyk) [inline src/elements/EditableVolumesRow/editableVolumesRow.scss:17]: you don't need these styles

File: src/elements/EditableVolumesRow/editableVolumesRow.scss [hunk 12]
```diff
@@ -14,5 +13,30 @@
...
+      max-width: 351px;
+      padding: 10px 20px 10px 0;
+    }
+
+    .volumes-access {
```
Comment (mariana-furyk) [inline src/elements/EditableVolumesRow/editableVolumesRow.scss:22]: you don't need these styles

File: src/elements/EditableVolumesRow/editableVolumesRow.scss [hunk 13]
```diff
@@ -14,5 +13,30 @@
...
+          padding: 20px 10px 10px 17px;
+
+          &__label {
+            left: 17px;
+          }
+        }
+      }
+    }
+
+    .volume-container,
```
Comment (mariana-furyk) [inline src/elements/EditableVolumesRow/editableVolumesRow.scss:37]: you don't need these styles

File: src/elements/JobsPanelTable/JobsPanelTable.js [hunk 14]
```diff
@@ -121,63 +121,56 @@ const JobsPanelTable = ({
...
+              className={`table__cell table__cell-${header
+                .split(' ')
+                .join('')
+                .toLowerCase()}`}
+              key={index}
+            >
               {header}
             </div>
           ))}
+          <div className="table__cell actions_cell"></div>
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTable/JobsPanelTable.js:134]: an empty tag should be self-closed

File: src/elements/JobsPanelTable/jobsPanelTable.scss [hunk 15]
```diff
@@ -50,70 +96,71 @@
     }
 
     .input-row-wrapper {
+      width: 100%;
+
       .input-wrapper {
         &:first-child {
-          flex: 0.4;
+          max-width: 351px;
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTable/jobsPanelTable.scss:103]: these styles breaks view

File: src/elements/JobsPanelTableRow/jobsPanelTableRow.scss [hunk 16]
```diff
@@ -2,20 +2,16 @@
...
     height: 55px;
     border-top: $tableBorder;
-    padding: 18px 20px;
 
     .actions_cell {
-      min-width: 36px;
-      position: absolute;
-      right: -8px;
-      top: 15px;
+      min-width: 44px;
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTableRow/jobsPanelTableRow.scss:13]: you can just add width and for flex max-width, but test if you really need to set width

File: src/elements/JobsPanelTableRow/jobsPanelTableRow.scss [hunk 17]
```diff
@@ -26,24 +22,49 @@
...
+    align-items: center;
+    padding: 10px;
     color: $primary;
-    margin-right: 5px;
+
+    &.add-input {
+      padding: 0;
+    }
+
+    &_disable {
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTableRow/jobsPanelTableRow.scss:47]: rename to disabled

File: src/common/TableActionsMenu/TableActionsMenu.js [hunk 18]
```diff
@@ -39,20 +39,25 @@ const TableActionsMenu = ({ item, menu, time }) => {
...
-              <span className="table-actions-container__icon">
-                {menuItem.icon}
-              </span>
-              {menuItem.label}
-            </div>
-          ))}
+          {menu.map(menuItem => {
+            if (menuItem.label === 'Remove' && visible) {
+              return undefined
+            }
```
Comment (mavdryk) [inline src/common/TableActionsMenu/TableActionsMenu.js:45]: That's not what I meant. I wrote to add a property visible to the action menu (the array of objects with menu items), but not in the props.

File: src/components/JobsPanel/panelData.json [hunk 19]
```diff
@@ -1,12 +1,23 @@
 {
   "data-inputs": {
-    "table-headers": ["Input Name", "Path"]
+    "table-headers": [
+      { "label": "Input Name", "classes": "inputName" },
+      { "label": "Path", "classes": "path" }
+    ]
```
Comment (mavdryk) [inline src/components/JobsPanel/panelData.json:6]: I believe you don't use `classes` in the code, you can delete them

File: src/components/JobsPanelDataInputs/JobsPanelDataInputs.js [hunk 20]
```diff
@@ -94,9 +102,17 @@ const JobsPanelDataInputs = ({
 
   const handleEditDataInput = () => {
     const currentInputs = { ...inputs }
+    currentInputs[selectedDataInput.data.name] = selectedDataInput.data.path
 
-    currentInputs[selectedDataInput.name] = selectedDataInput.path
+    let currentDataInput = [...defaultDataInputs]
+    let currentDataInputIndex = defaultDataInputs.findIndex(
```
Comment (mariana-furyk) [inline src/components/JobsPanelDataInputs/JobsPanelDataInputs.js:108]: change let to const

File: src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js [hunk 21]
```diff
@@ -63,7 +63,7 @@ const JobsPanelDataInputsView = ({
           setSelectedDataInput={setSelectedDataInput}
         >
           {addNewInput ? (
-            <>
+            <div className="table__row-addItem">
```
Comment (mariana-furyk) [inline src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js:66]: don't use camelCase for css classes

File: src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js [hunk 22]
```diff
@@ -109,74 +109,80 @@ const JobsPanelDataInputsView = ({
...
+                    }
+                    label="Path"
+                    className="input-row__item"
+                    floatingLabel
+                    type="text"
+                  />
+                </div>
+                <div
+                  className={`input-row-wrapper ${newVolume.type === 'V3IO' &&
+                    ' no-border'}`}
```
Comment (mavdryk) [inline src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js:136]: delete space before `no-border`

File: src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js [hunk 23]
```diff
@@ -109,74 +109,80 @@ const JobsPanelDataInputsView = ({
...
+                    onClick={type =>
                       setNewVolume({
                         ...newVolume,
-                        resourcesPath: resourcesPath
+                        type: find(selectOptions.volumeType, ['id', type]).id
                       })
                     }
-                    label="Resource path"
+                    options={selectOptions.volumeType}
+                    label={newVolume.type.length > 0 ? newVolume.type : 'Type'}
```
Comment (mariana-furyk) [inline src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js:146]:
```suggestion
                    label={newVolume.type.length ? newVolume.type : 'Type'}
```

File: src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js [hunk 24]
```diff
@@ -109,74 +109,80 @@ const JobsPanelDataInputsView = ({
...
                 </div>
-              )}
+                {newVolume.type === 'V3IO' && (
+                  <div className="input-row-wrapper">
+                    <Input
+                      onChange={accessKey =>
+                        setNewVolume({ ...newVolume, accessKey: accessKey })
+                      }
+                      label="Access Key"
+                      className="input-row__item volume-access-key"
```
Comment (mariana-furyk) [inline src/components/JobsPanelDataInputs/JobsPanelDataInputsView.js:167]: you don't use class volume-access-key

File: src/components/JobsPanelParameters/JobsPanelParameters.js [hunk 25]
```diff
@@ -61,7 +65,10 @@ const JobsPanelParameters = ({
       return setAddNewParameter(false)
     }
 
+    let defaultParameters = parseDefaultContent(parametersArray)
```
Comment (mariana-furyk) [inline src/components/JobsPanelParameters/JobsPanelParameters.js:68]: change let to const

File: src/elements/EditableDataInputsRow/EditableDataInputsRow.js [hunk 26]
```diff
@@ -12,20 +12,25 @@ const EditableDataInputsRow = ({
 }) => {
   return (
     <div className="table__row edit-row">
-      <div className="table__cell">{selectedDataInput.name}</div>
-      <div className="table__cell">
+      <div className="table__cell table__cell-name">
```
Comment (mariana-furyk) [inline src/elements/EditableDataInputsRow/EditableDataInputsRow.js:15]: you don't add any style for table__cell-name, so I think you don't need it

File: src/elements/EditableDataInputsRow/EditableDataInputsRow.js [hunk 27]
```diff
@@ -12,20 +12,25 @@ const EditableDataInputsRow = ({
...
+              data: { ...selectedDataInput.data, path: path }
+            })
           }
           type="text"
-          value={selectedDataInput.path}
+          value={selectedDataInput.data.path}
         />
       </div>
-      <div className="table__cell">
+      <div className="table__cell actions_cell">
```
Comment (mariana-furyk) [inline src/elements/EditableDataInputsRow/EditableDataInputsRow.js:30]: rename this class name, cell is not modificator

File: src/elements/JobsPanelTableRow/JobsPanelTableRow.js [hunk 28]
```diff
@@ -5,34 +5,65 @@ import Tooltip from '../../common/Tooltip/Tooltip'
...
       <div className="table__cell actions_cell">
-        <TableActionsMenu menu={actionsMenu} item={item} />
+        {((!item.isValueEmpty && !item.isDefault) ||
+          (item.isValueEmpty && item.isDefault)) && (
+          <TableActionsMenu
+            item={item}
+            menu={actionsMenu}
+            visible={
+              (!item.data.isValueEmpty || !item.data.path) &&
+              item.isDefault === true
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTableRow/JobsPanelTableRow.js:55]:
```suggestion
              item.isDefault
```

Comment (mavdryk) [inline src/elements/JobsPanelTableRow/JobsPanelTableRow.js:55]: never compare with `true`. It's redundant logic

File: src/elements/JobsPanelTable/JobsPanelTable.js [hunk 29]
```diff
@@ -30,89 +30,113 @@ const JobsPanelTable = ({
...
+          onClick: item => {
+            handleDelete(item)
+          }
+        }
+      ]
+    },
+    [handleEdit, handleDelete]
+  )
 
   return (
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTable/JobsPanelTable.js:141]: Remove html to view component

File: src/elements/JobsPanelTableRow/JobsPanelTableRow.js [hunk 30]
```diff
@@ -5,34 +5,60 @@ import Tooltip from '../../common/Tooltip/Tooltip'
...
-const JobsPanelTableRow = ({ actionsMenu, item, row }) => {
+const JobsPanelTableRow = ({ actionsMenu, item }) => {
   return (
     <div className="table__row">
-      {row.map((cell, i) => {
+      {map(item.data, (value, property) => {
         return (
-          <div className="table__cell" key={i + cell}>
+          <div
+            className={classNames({
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTableRow/JobsPanelTableRow.js:21]: declare variable for class names

File: src/elements/JobsPanelTableRow/jobsPanelTableRow.scss [hunk 31]
```diff
@@ -2,48 +2,64 @@
...
           display: block;
 
           &__body {
-            bottom: -102px;
             right: 22px;
           }
         }
       }
     }
+
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTableRow/jobsPanelTableRow.scss:31]: you don't use this class

File: src/elements/JobsPanelTitle/JobsPanelTitleView.js [hunk 32]
```diff
@@ -23,10 +25,11 @@ const JobsPanelTitleView = ({
   return (
     <div className="job-panel__title">
       <div
-        className={`job-panel__title-wrapper ${isEdit &&
-          'job-panel__title-wrapper_edit'} ${!openScheduleJob &&
-          !isEdit &&
-          'job-panel__title-wrapper_hover'}`}
+        className={classNames({
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTitle/JobsPanelTitleView.js:28]: declare variable for class names

File: src/scss/main.scss [hunk 33]
```diff
@@ -148,6 +149,8 @@ input {
 }
 
 .data-ellipsis {
+  min-width: 100%;
```
Comment (mariana-furyk) [inline src/scss/main.scss:152]: do not add width to class here, add them in styles where you need it

File: src/elements/JobsPanelTable/jobsPanelTable.scss [hunk 34]
```diff
@@ -80,12 +80,12 @@
 
     .table__cell {
       &_edit {
-        padding: 10px 20px 10px 0;
+        min-width: 175.5px;
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTable/jobsPanelTable.scss:83]: don't add floating values

File: src/elements/JobsPanelTable/jobsPanelTable.scss [hunk 35]
```diff
@@ -80,12 +80,12 @@
...
+        padding: 10px 0 10px 0;
```
Comment (mariana-furyk) [inline src/elements/JobsPanelTable/jobsPanelTable.scss:84]:
```suggestion
        padding: 10px 0;
```

**Pull request closed without merging**